crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -412,23 +373,27 @@ where
             })
             .collect::<HashMap<_, _>>();
 
-        caches.iter_mut().for_each(|state| {
-            // Filter-out piece indices that are stored, but should not be as well as clean
-            // `inserted_piece_indices` from already stored piece indices, leaving just those that are
-            // still missing in cache
-            state
-                .stored_pieces
-                .extract_if(|key, _offset| piece_indices_to_store.remove(key).is_none())
-                .for_each(|(_piece_index, offset)| {
-                    state.free_offsets.push_front(offset);
-                });
-        });
+        let mut piece_caches_capacity_used = vec![0i64; caches.backends.len()];
```

Comment:
Why would capacity possibly be negative?

If we need to use it for metrics, cast to `i64` closer to metrics (the only reason we don't use `u64` there is that gauges do not like `u64` for some reason and I didn't want to create custom data structures just for metrics purposes to support `u64`).

Reply:
You're right, I guess I should have used u32 and cast to i64 closer to metrics